### PR TITLE
Fix testcase stub of `createModelDataFor`

### PR DESCRIPTION
**Pull request by machty**
`createModelDataFor` was the old name before it was changed
to `createRecordDataFor`

### CODE REVIEW

**Comment by igorT** [comment]
@machty can you please add an assert in the code we are stubbing to ensure this time it is getting called.

**Comment by machty** [comment]
@igorT while addressing your feedback I noticed these tests had an identical setup and what appeared to be a lot of leftover cruft from the original RFC implementation. It's much smaller now, performs the same tests, and the stub assertion you asked for is accomplished just by testing that that the constructed RecordData for a model has the expected type (which is what one of the tests was doing before).

File: packages/-ember-data/tests/integration/record-data/store-compatibility-test.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  test(`store.unloadRecord on a record with default RecordData with relationship to a record with custom RecordData does not error`, async function (assert) {
+    run(() => chris.unloadRecord());
+  });
+
+  test(`store.unloadRecord on a record with custom RecordData with relationship to a record with default RecordData does not error`, async function (assert) {
+    run(() => shen.unloadRecord());
+  });
+});
+
+module('RecordData - Instantiation', function (hooks) {
```
Comment (igorT) [inline packages/-ember-data/tests/integration/record-data/store-compatibility-test.js:99]: Can we not split the two modules into two files please?

**Comment by igorT** [comment]
@machty great work!

**Comment by runspired** [comment]
@machty if you hapeen to have time to rebase this I think the tests will pass and I'd be happy to merge

**Comment by runspired** [comment]
something is odd here, the more recent commits clearly show test refactoring that the github diff cannot seem to see :/

**Pull request closed without merging**